src/main/java/net/dv8tion/jda/api/hooks/VoiceDispatchInterceptor.java

```diff
@@ -0,0 +1,246 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.hooks;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildVoiceState;
+import net.dv8tion.jda.api.entities.VoiceChannel;
+import net.dv8tion.jda.api.managers.DirectAudioController;
+import org.json.JSONObject;
+import org.json.JSONString;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Interceptor used to handle critical voice dispatches.
+ * <br>This will make it impossible to connect to voice channels with
+ * the built-in {@link net.dv8tion.jda.api.managers.AudioManager AudioManager}.
+ * It is expected that the user has some other means of establishing voice connections when this is used.
+ */
+public interface VoiceDispatchInterceptor
+{
+    /**
+     * Handles the <b>VOICE_SERVER_UPDATE</b>.
+     *
+     * @param update
+     *        The {@link VoiceServerUpdate} to handle
+     */
+    void interceptServerUpdate(@Nonnull VoiceServerUpdate update);
+
+    /**
+     * Handles the <b>VOICE_STATE_UPDATE</b>.
+     * <br>This indicates the user might have moved to a new voice channel.
+     *
+     * @param  update
+     *         The {@link VoiceStateUpdate} to handle
+     *
+     * @return True, if a connection was previously established
+     */
+    boolean interceptStateUpdate(@Nonnull VoiceStateUpdate update);
+
+    /**
+     * Abstraction for all relevant voice updates
+     *
+     * @see VoiceServerUpdate
+     * @see VoiceStateUpdate
+     */
+    interface VoiceUpdate extends JSONString
+    {
+        /**
+         * The {@link Guild} for this update
+         *
+         * @return The guild
+         */
+        Guild getGuild();
+
+        /**
+         * The raw JSON object that was parsed from this update
+         *
+         * @return The raw JSON object
+         */
+        JSONObject getJSON();
+
+        @Override
+        default String toJSONString()
+        {
+            return getJSON().toString();
+        }
+
+        /**
+         * Shortcut to access the audio controller of this JDA instance
+         *
+         * @return The {@link DirectAudioController} for this JDA instance
+         */
+        default DirectAudioController getAudioController()
+        {
+            return getJDA().getAudioController();
+        }
+
+        /**
+         * Shortcut to access the guild id
+         *
+         * @return The guild id
+         */
+        default long getGuildIdLong()
```

Comment:
I feel like this and the `getGuildId` shortcuts aren't really worth it.

Reply:
Worth it? I don't feel like they hurt by being there